sphinx_automodapi/smart_resolver.py

```diff
@@ -13,22 +13,58 @@
 corrected if possible.
 """
 
+import os
+import glob
+import uuid
+import tempfile
 from docutils.nodes import literal, reference
 
+# NOTE: This extension originally used a dictionary stored in app.env to keep
```

Comment:
This whole approach seems a bit problematic, as it depends on file systems and sort of implementation details.  Is there a reason you didn't try using the`event-env-merge-info` event to do the parallel gathering (http://www.sphinx-doc.org/en/stable/extdev/appapi.html#event-env-merge-info)?  I thought it was meant to solve exactly this problem? (Although I've not used it myself...)